festim/hydrogen_transport_problem.py

```diff
@@ -123,11 +129,50 @@ def initialise(self):
         self.t = fem.Constant(self.mesh.mesh, 0.0)
         self.dt = self.settings.stepsize.get_dt(self.mesh.mesh)
 
+        self.define_temperature()
         self.define_boundary_conditions()
         self.create_formulation()
         self.create_solver()
         self.defing_export_writers()
 
+    def define_temperature(self):
```

Comment:
Could we add a test at the beginning of this method that checks if `self.temperature` is `None` and raises an error if that's the case?